ui/components/multichain/network-list-item/network-list-item.js

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import Box from '../../ui/box/box';
+import {
+  AlignItems,
+  IconColor,
+  BorderRadius,
+  Color,
+  Size,
+  JustifyContent,
+} from '../../../helpers/constants/design-system';
+import {
+  AvatarNetwork,
+  ButtonIcon,
+  ICON_NAMES,
+  Text,
+} from '../../component-library';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+
+export const NetworkListItem = ({
+  name,
+  iconSrc,
+  selected = false,
+  onClick,
+  onDeleteClick,
+}) => {
+  const t = useI18nContext();
+  return (
+    <Box
+      onClick={onClick}
+      padding={4}
+      gap={2}
+      backgroundColor={selected ? Color.primaryMuted : Color.transparent}
+      className={classnames('network-list-item', {
+        'network-list-item--selected': selected,
+      })}
+      as="button"
+      alignItems={AlignItems.center}
+      justifyContent={JustifyContent.spaceBetween}
+    >
+      {selected && (
+        <Box
+          className="network-list-item__selected-indicator"
+          borderRadius={BorderRadius.pill}
+          backgroundColor={Color.primaryDefault}
+        />
+      )}
+      <AvatarNetwork name={name} src={iconSrc} />
+      <Box className="network-list-item__network-name">
+        <Text as="div" ellipsis>
```

Comment:
Why `as="div"` does "span" not work?

Reply:
The ellipsis functionality breaks when there's a tooltip via `span`

Reply:
<img width="530" alt="NoEllipsis" src="https://user-images.githubusercontent.com/46655/228302559-3b889152-91a4-4b7d-bb2f-bb438852d22e.png">

Reply:
Roger that.